Stops random maintenance spawners from being able to spawn ashes in space

vinylspiders opened the pull request:
## About The Pull Request

True to their name, 'maintenance spawners' have been spawning more work for maintainers for the past few years with this annoying runtime. It is rng whether or not it occurs, but it happens frequently enough..and sometimes in multiple tests if you are lucky!

![firefox_Fokh0Dehp3](https://github.com/user-attachments/assets/67224048-d08c-46e2-a404-924dba71ef34)

See: a triple whammy

![image](https://github.com/user-attachments/assets/6fe84969-c952-4f8a-bd9a-4b051a53e0a0)

Finally, inspired by this recent event I have made it my mission to put an end to this for good.

This pr:

1) Prevents mappers from placing `/obj/effect/spawner/random/maintenance` in space or openspace turfs, insisting that they must use the safe `/obj/effect/spawner/random/maintenance/no_decals` subtype instead which does not spawn ashes.

2) If somehow this ends up on a space or openspace turf outside of mapload, we will just delete the offending spawner and spawn the `/obj/effect/spawner/random/maintenance/no_decals` in place. There's literally no reason we should be wasting any more of our maintainers' time with this!

Hopefully this puts this runtime into the crematorium for good because it's stupid.

![image](https://github.com/user-attachments/assets/3612de52-1604-4f83-a217-cdae9f22e6c4)

## Why It's Good For The Game

No more rng garbage in the CI runs.

## Changelog

:cl:
fix: random maintenance spawners can no longer end up in space, spawn ashes, and annoy maintainers.
/:cl:

vinylspiders committed:
Stops random maintenance spawners from being able to be placed in space

timothymtorres commented:
There are some trash random spawners that spawn objects in a certain radius around the spawner itself. (most of them are 3x3, but some can be larger like 5x5) So if any of these spawners are placed near a window, you can end up with trash in space.

FlufflesTheDog commented:
> > There are some trash random spawners that spawn objects in a certain radius around the spawner itself. (most of them are 3x3, but some can be larger like 5x5) So if any of these spawners are placed near a window, you can end up with trash in space.
> 
> That is quite silly...those should probably be blocked by walls

I actually had a fun time troubleshooting this exact thing

A random crate spawner spawned a dumpster which spawned a garbage spawner which spawned a cockroach 2 tiles away in space. The kicker? Windows do block the "two tiles away" thing, but window *spawners* do not :)

vinylspiders committed:
Adds ruin coords to logging

vinylspiders committed:
Revert "Adds ruin coords to logging"

This reverts commit 526947ac8e3d106bc2c258e9e809d65524a8ffac.

vinylspiders committed:
Use stack_trace instead

Turns out it doesn't actually give a map log failure if the mapping error occurs in a ruin (!??!)

vinylspiders committed:
Let no_decal type pass

vinylspiders committed:
Update maintenance.dm

SmArtKar commented on code/game/objects/effects/spawners/random/maintenance.dm:
```diff
@@ -19,6 +19,13 @@
+		if(mapload)
+			stack_trace("[src] at [AREACOORD(src)] tried to spawn in a space or openspace tile! Mappers please use /obj/effect/spawner/random/maintenance/no_decals instead! Note: the coords might be for a ruin so keep that in mind.")
+		else
+			new /obj/effect/spawner/random/maintenance/no_decals(loc) // if this happens outside of mapload, just spawn a no_decals version and move on
+		return INITIALIZE_HINT_QDEL
```
This breaks spawn amount for subtypes, better to runtime on mapload and check for turf validity in can_spawn

vinylspiders committed:
Fixes losing the spawner amounts

timothymtorres commented on code/game/objects/effects/spawners/random/maintenance.dm:
```diff
@@ -23,7 +23,12 @@
 			stack_trace("[src] at [AREACOORD(src)] tried to spawn in a space or openspace tile! Mappers please use /obj/effect/spawner/random/maintenance/no_decals instead! Note: the coords might be for a ruin so keep that in mind.")
 		else
-			new /obj/effect/spawner/random/maintenance/no_decals(loc) // if this happens outside of mapload, just spawn a no_decals version and move on
+			// if this happens outside of mapload, just spawn a no_decals version and move on
+			var/regex/no_decal_regex = regex("maintenance")
+			var/new_path_string = no_decal_regex.Replace("[type]", "maintenance/no_decals")
+			var/no_decals_type = text2path(new_path_string)
+			new no_decals_type(loc)
```
why not just use an `istype(type, /obj/effect/spawner/random/maintenance)` check and spawn the no_decals version? What is the purpose of doing a regex check here? Are there several `no_decal` subtypes?

SmArtKar replied:
Any particular reason to do this path thing instead of just removing decals from own pool and continuing init?
